pkg/migration/migrations/separated_intents.go

```diff
@@ -301,17 +308,35 @@ func (m *migrateLockTablePool) run(ctx context.Context, workerIdx int) {
 			handleError(err)
 			continue
 		} else {
-			retryRequest = nextReq
 			if nextReq == nil {
 				// This range has been migrated.
 				atomic.AddUint64(&m.finished, 1)
+				m.mu.Lock()
+				m.mu.lowWater[workerIdx] = retryRequest.rangeStart
+				m.mu.Unlock()
 			}
```

Comment:
You can do a little better in the case of a nextReq with a resume span (a rare but still possible case):

```suggestion
} else if nextReq.barrierDone {
 // could use a better bool name to denote that this was a resumeSpan
 m.mu.Lock()
 m.mu.lowWater[workerIdx] = retryRequest.start
 m.mu.Unlock()
}
```